router/core/transport.go

```diff
@@ -116,6 +117,13 @@ func (ct *CustomTransport) RoundTrip(req *http.Request) (resp *http.Response, er
 
 	if !ct.allowSingleFlight(req) {
 		resp, err = ct.roundTripper.RoundTrip(req)
+		if err == nil && ct.isUpgradeError(req, resp) {
```

Comment:
Why do we apply this conditionally for allowSingleFlight == false?

Reply:
we deactivate SF for Upgrade Requests, so if it's not SF, it's not an upgrade request